package/MDAnalysis/topology/LAMMPSParser.py

```diff
@@ -627,33 +640,76 @@ def parse(self, **kwargs):
             fin.readline()  # y
             fin.readline()  # z
 
-            indices = np.zeros(natoms, dtype=int)
-            types = np.zeros(natoms, dtype=object)
 
+     
+            # Next line contains the headers for the atom data
             atomline = fin.readline()  # ITEM ATOMS
             attrs = atomline.split()[2:]  # attributes on coordinate line
             col_ids = {attr: i for i, attr in enumerate(attrs)}  # column ids
+            headers = {header: header in col_ids for header in DUMP_HEADERS} # Find the headers present in the dump file
+            atom_data = {}
+            # Initialize the atom data arrays
+            if headers["id"]:
+                atom_data["id"] = np.zeros(natoms, dtype=np.int32)
+            else:
+                raise ValueError("No id column found in dump file")
+            if headers["type"]:
+                atom_data["type"] = np.zeros(natoms, dtype=object)
+            else:
+                warnings.warn("Set all atom types to 1")
+                atom_data["type"] = np.ones(natoms, dtype=object)
+            if headers["mass"]:
+                atom_data["mass"] = np.zeros(natoms, dtype=np.float64)
+            if headers["mol"]:
+                atom_data["mol"] = np.zeros(natoms, dtype=int)
+            else:
+                atom_data["mol"] = np.ones(natoms, dtype=int) # Set resids to 1 if not present
+            if headers["q"]:
+                atom_data["q"] = np.zeros(natoms, dtype=np.float32)
+            if headers["element"]:
+                atom_data["element"] = np.zeros(natoms, dtype=object)
+
 
             for i in range(natoms):
                 fields = fin.readline().split()
+                for header in DUMP_HEADERS:
+                    if headers[header]:
+                        atom_data[header][i] = fields[col_ids[header]]
```

Comment:
Blocking: If not this solution another one
```suggestion
                for header, col_id in col_ids.items():
                    atom_data[header][i] = fields[col_id]
```